azurerm_cosmosdb_account: ip_range_filter doesnt work any longer

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.9.3

### AzureRM Provider Version

4.0.0

### Affected Resource(s)/Data Source(s)

azurerm_cosmosdb_account

### Terraform Configuration Files

```hcl
resource "azurerm_cosmosdb_account" "main" {
  name                = var.name
  resource_group_name = var.resource_group_name
  location            = var.location
  offer_type          = var.offer_type

  create_mode                           = var.create_mode
  default_identity_type                 = var.default_identity_type
  kind                                  = var.kind
  ip_range_filter                       = var.ip_range_filter
  free_tier_enabled                     = var.free_tier_enabled
  analytical_storage_enabled            = var.analytical_storage_enabled
  automatic_failover_enabled            = var.automatic_failover_enabled
  public_network_access_enabled         = var.public_network_access_enabled
  is_virtual_network_filter_enabled     = var.is_virtual_network_filter_enabled
  key_vault_key_id                      = var.key_vault_key_id
  multiple_write_locations_enabled      = var.multiple_write_locations_enabled
  access_key_metadata_writes_enabled    = var.access_key_metadata_writes_enabled
  mongo_server_version                  = var.mongo_server_version
  network_acl_bypass_for_azure_services = var.network_acl_bypass_for_azure_services
  network_acl_bypass_ids                = var.network_acl_bypass_ids
  local_authentication_disabled         = var.local_authentication_disabled

  consistency_policy {
    consistency_level       = var.consistency_policy.consistency_level
    max_interval_in_seconds = try(var.consistency_policy.max_interval_in_seconds, null)
    max_staleness_prefix    = try(var.consistency_policy.max_staleness_prefix, null)
  }

  dynamic "geo_location" {
    for_each = var.geo_location
    content {
      location          = geo_location.value.location
      failover_priority = geo_location.value.failover_priority
      zone_redundant    = geo_location.value.zone_redundant
    }
  }

  dynamic "capabilities" {
    for_each = var.capabilities

    content {
      name = capabilities.value
    }
  }

  dynamic "virtual_network_rule" {
    for_each = var.virtual_network_rule == null ? [] : [1]

    content {
      id                                   = var.virtual_network_rule.id
      ignore_missing_vnet_service_endpoint = var.virtual_network_rule.ignore_missing_vnet_service_endpoint
    }
  }

  dynamic "backup" {
    for_each = var.backup == null ? [] : [1]

    content {
      type                = var.backup.type
      interval_in_minutes = var.backup.interval_in_minutes
      retention_in_hours  = var.backup.retention_in_hours
      storage_redundancy  = var.backup.storage_redundancy
    }
  }

  dynamic "cors_rule" {
    for_each = var.cors_rule == null ? [] : [1]

    content {
      allowed_headers    = var.cors_rule.allowed_headers
      allowed_methods    = var.cors_rule.allowed_methods
      allowed_origins    = var.cors_rule.allowed_origins
      exposed_headers    = var.cors_rule.exposed_headers
      max_age_in_seconds = var.cors_rule.max_age_in_seconds
    }
  }

  dynamic "identity" {
    for_each = var.identity == null ? [] : [1]

    content {
      type = var.identity.type
    }
  }

  dynamic "restore" {
    for_each = var.restore == null ? [] : [1]

    content {
      source_cosmosdb_account_id = var.restore.source_cosmosdb_account_id
      restore_timestamp_in_utc   = var.restore.restore_timestamp_in_utc

      database {
        name             = var.restore.database.name
        collection_names = var.restore.database.collection_names
      }
    }
  }

  dynamic "analytical_storage" {
    for_each = var.analytical_storage == null ? [] : [1]

    content {
      schema_type = var.analytical_storage.schema_type
    }
  }

  dynamic "capacity" {
    for_each = var.capacity == null ? [] : [1]

    content {
      total_throughput_limit = var.capacity.total_throughput_limit
    }
  }

  tags = var.tags
}

variable "ip_range_filter" {
  type = string
  default = "x.y.z.1,x.y.z.2,x.y.z.3"
}
```


### Debug Output/Panic Output

```shell
Error: -23T03:44:10.893Z [ERROR] vertex "module.platform.module.cosmosdb[\"main\"].azurerm_cosmosdb_account.main" error: missing expected [
Error: -23T03:44:10.893Z [ERROR] vertex "module.platform.module.cosmosdb.azurerm_cosmosdb_account.main (expand)" error: missing expected [
```


### Expected Behaviour

Es per documentation (wasn't changed neither before 4.0.0 nor after):

`[ip_range_filter](https://registry.terraform.io/providers/hashicorp/azurerm/3.116.0/docs/resources/cosmosdb_account#ip_range_filter) - (Optional) CosmosDB Firewall Support: This value specifies the set of IP addresses or IP address ranges in CIDR form to be included as the allowed list of client IPs for a given database account. IP addresses/ranges must be comma separated and must not contain any spaces.`


### Actual Behaviour

before 4.0.0:
it works as described in `Terraform Configuration Files`.
after 4.0.0:
it doesn't work, even if I set thisv ariable up to set type or even to list. 
For `string` it started to fail on type error.
For `set` or `list` it started to fail on `error: missing expected [`


### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/website/docs/guides/4.0-upgrade-guide.html.markdown
+++ b/website/docs/guides/4.0-upgrade-guide.html.markdown
@@ -703,6 +703,7 @@ The deprecated data source has been superseded by `azurerm_mssql_server` and has
 * The deprecated `enable_free_tier` property has been removed in favour of the `free_tier_enabled` property.
 * The deprecated `enable_automatic_failover` property has been removed in favour of the `automatic_failover_enabled` property.
 * The `minimal_tls_version` property now defaults to `Tls12`.
+* The `ip_range_filter` property is now a set that will only accept valid CIDR values.
 
 ### `azurerm_cosmosdb_sql_container`
 

--- a/website/docs/r/cosmosdb_account.html.markdown
+++ b/website/docs/r/cosmosdb_account.html.markdown
@@ -136,7 +136,7 @@ The following arguments are supported:
 
 * `geo_location` - (Required) Specifies a `geo_location` resource, used to define where data should be replicated with the `failover_priority` 0 specifying the primary location. Value is a `geo_location` block as defined below.
 
-* `ip_range_filter` - (Optional) CosmosDB Firewall Support: This value specifies the set of IP addresses or IP address ranges in CIDR form to be included as the allowed list of client IPs for a given database account. IP addresses/ranges must be comma separated and must not contain any spaces.
+* `ip_range_filter` - (Optional) A set of IP addresses or IP address ranges in CIDR form to be included as the allowed list of client IPs for a given database account. For example `["55.0.1.0/24", "55.0.2.0/24"]`.
 
 ~> **Note:** To enable the "Allow access from the Azure portal" behavior, you should add the IP addresses provided by the [documentation](https://docs.microsoft.com/azure/cosmos-db/how-to-configure-firewall#allow-requests-from-the-azure-portal) to this list.
 

